lib/view_model/set_up_2fa_viewmodel.dart

```diff
@@ -84,24 +85,17 @@ abstract class Setup2FAViewModelBase with Store {
   bool get shouldRequireTOTP2FAForAllSecurityAndBackupSettings =>
       _settingsStore.shouldRequireTOTP2FAForAllSecurityAndBackupSettings;
 
-  void _getRandomBase32SecretKey() {
-    final randomBase32Key = Utils.generateRandomBase32SecretKey(16);
-    _setBase32SecretKey(randomBase32Key);
-  }
-
   @action
-  void setUseTOTP2FA(bool value) {
-    _settingsStore.useTOTP2FA = value;
-  }
+  void generateSecretKey() {
+    totpSecretKey = Utils.generateRandomBase32SecretKey(16);
```

Comment:
you are only setting the `totpSecretKey` here, and this function is only called in the setup screen, so when if a user setup the 2FA and then close the app and then re-open it, the `totpSecretKey` won't get initialized, thus it will fail to verify the code that the user enters